xdsl/ir.py

```diff
@@ -958,16 +965,40 @@ def move_blocks(self, region: Region) -> None:
         for block in region.blocks:
             block.parent = region
 
-    def get_toplevel_object(self) -> Operation | Block | Region:
-        """Get the operation, block, or region ancestor that has no parents."""
-        if self.parent is None:
-            return self
-        return self.parent.get_toplevel_object()
 
-    def is_ancestor(self, op: Operation | Block | Region) -> bool:
-        "Returns true if the region is an ancestor of the operation, block, or region."
-        if op is self:
-            return True
-        if op.parent is None:
-            return False
-        return self.is_ancestor(op.parent)
+class DiagnosticException(Exception):
```

Comment:
Why did you decide to move this here? Circular imports?

Reply:
Yes, I moved this here or circular imports. Of course, it is not resolved. Any ideas to make it better?